app/src/main/java/org/systers/mentorship/view/fragments/TasksFragment.kt

```diff
@@ -0,0 +1,74 @@
+package org.systers.mentorship.view.fragments
+
+import android.app.AlertDialog
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Bundle
+import android.support.v7.widget.LinearLayoutManager
+import android.widget.EditText
+import kotlinx.android.synthetic.main.fragment_mentorship_tasks.*
+import kotlinx.android.synthetic.main.task_list_item.*
+import org.systers.mentorship.MentorshipApplication
+import org.systers.mentorship.R
+import org.systers.mentorship.view.activities.MainActivity
+import org.systers.mentorship.view.adapters.TasksAdapter
+import org.systers.mentorship.viewmodels.TasksViewModel
+
+/**
+ * The fragment is responsible for showing the all mentorship tasks
+ * and achievements. It also allows to add new tasks.
+ */
+class TasksFragment : BaseFragment() {
+
+    companion object {
+        /**
+         * Creates an instance of [TasksFragment]
+         */
+        fun newInstance() = TasksFragment()
+        val TAG = TasksFragment::class.java.simpleName
+    }
+
+    private lateinit var taskViewModel: TasksViewModel
+
+    override fun getLayoutResourceId(): Int = R.layout.fragment_mentorship_tasks;
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        taskViewModel = ViewModelProviders.of(this).get(TasksViewModel::class.java)
+        taskViewModel.getTasks()
+
+        rvTasks.apply {
+            layoutManager = LinearLayoutManager(context)
+            adapter = TasksAdapter(taskViewModel.taskList, markTask)
+        }
+
+        ivAddItem.setOnClickListener {
+            showDialog()
+        }
+    }
+
+    /**
+     * The function creates a dialog box through whoch new tasks can be added
+     */
+    fun showDialog() {
+        val builder = AlertDialog.Builder(context)
+        val inflater = layoutInflater
+        builder.setTitle(MentorshipApplication.getContext().getString(R.string.add_new_task))
```

Comment:
Could you pass `MentorshipApplication.getContext()``to a variable, since you're calling it 3 times in this function? @Rimjhim28

Reply:
@isabelcosta Done! :D